### PR TITLE
Delay slider fetch place

### LINKED ISSUE #13: Use delay slider setting to control frequency of place fetches
Early in the project, we set things up so the app would not try to fetch new places again before the previous call had completed. This story extends this functionality.

AC:

1. App should wait to fetch new places until after the previous call has completed plus the number of seconds the user has set in the delay slider

### PR DESCRIPTION
**Pull request by AshlingH**
## Description
For this issue, we applied a timeout to our existing handlecenterchange function so the app waits to make the next api call and fetching the next pins.The user can move the slider in the settings modal to determine the time between each call.

## Related Issue

#13 

## Acceptance Criteria

- [x] App should wait to fetch new places until after the previous call has completed plus the number of seconds the user has set in the delay slider

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

### CODE REVIEW

File: src/components/Map.js [hunk 1]
```diff
@@ -27,6 +26,8 @@ const Map = ({
     ).then((res) => setNearbyPlaces(res));
   }, [mapProperties.center.lat, mapProperties.center.lng, setNearbyPlaces]);
 
+  // converts value in seconds to be milliseconds for the setTimeout
+  const fetchDelay = localStorage.getItem("sliderValueInLocalStorage") * 1000;
```
Comment (segdeha) [inline src/components/Map.js:30]: Way to remember to convert the units for the delay! Unlike someone who is currently reviewing your PR… 😅

File: src/components/Map.js [hunk 2]
```diff
@@ -36,17 +37,21 @@ const Map = ({
       const panLat = panCoords.lat();
       const panLng = panCoords.lng();
       setUserHasPanned(true);
-      handleCenterChanged(panLat, panLng);
+      setTimeout(() => {
```
Comment (segdeha) [inline src/components/Map.js:40]: So this should work well for the basic case, but let's think through a scenario:

1. Imma User has her delay set to 30 seconds
2. She pans the map
3. 20 seconds later, she pans the map again

What I _think_ would happen is Imma will be waiting most of a minute to see new pins. Imagine she continues to pan, what will happen is we will continue to queue up calls which will then fire 30 seconds after each call to this function.

`setTimeout` returns an ID you can use to `cancelTimeout` so consider stashing that in local state and canceling before you make a new `setTimeout` call.

**Review by segdeha — APPROVED** [review]
I left 1 piece of functional feedback. I'm approving because I'd easily be convinced that we can merge this as an enhancement to the app then address my feedback as a future improvement, but if you want to tackle it now go for it!

**Review by jendevelops — APPROVED** [review]
I agree with Andrew that we can just address this as a future issue, so we don't get blocked by improving the setTimeout. @AshlingH @mjhughes707 could one of you add this as an enhancement issue on our issues page? Other than that, great and simple code. Easy to follow. Two thumbs way up!

File: src/components/Map.js (hunk 1 above)
Comment (mjhcodes) [reply src/components/Map.js:30]: I wish I could say that I remembered this on the 1st try. 😬

**Pull request merged**